composables/use-paywall-modal.ts

```diff
@@ -0,0 +1,121 @@
+import { PaywallModal } from '#components'
+
+export function usePayWall({
+  isFullScreen = false,
+  dismissible = true,
+  originalYearlyPrice,
+  originalMonthlyPrice,
+  discountedYearlyPrice,
+  discountedMonthlyPrice,
+  onClose,
+}: {
+  isFullScreen?: boolean
+  dismissible?: boolean
+  originalYearlyPrice?: string | number
+  originalMonthlyPrice?: string | number
+  discountedYearlyPrice?: string | number
+  discountedMonthlyPrice?: string | number
+  onClose?: () => void
+} = {}) {
+  const { t: $t } = useI18n()
+  const selectedPlan = ref('yearly')
+  const { loggedIn: hasLoggedIn } = useUserSession()
+  const accountStore = useAccountStore()
+  const isProcessingSubscription = ref(false)
+  const { handleError } = useErrorHandler()
+  const localeRoute = useLocaleRoute()
+  const toast = useToast()
+
+  const isLikerPlus = computed(() => {
+    if (!hasLoggedIn.value) return false
+    const { user } = useUserSession()
```

Comment:
can move outside the computed function and place next to composables in the top
```suggestion
```

Reply:
[🎨 Update paywall functionality to useSubscription and method names](https://github.com/likecoin/3ook-com/pull/57/commits/ce9dceb62e155e58725fe701d112848332072250)